vars/uploadToS3.groovy

```diff
@@ -0,0 +1,7 @@
+void call(Map args = [:]){
+
+    withAWS(role: "${ARTIFACT_UPLOAD_ROLE_NAME}", roleAccount: "${AWS_ACCOUNT_PUBLIC}", duration: 900, roleSessionName: 'jenkins-session') {
```

Comment:
Why put this few lines into its own library/method?
Its gonna be confusing if we want to switch roles and accounts and upload to s3. 
Is this really required?

Reply:
Added this in response to @dblock's review [comment](https://github.com/opensearch-project/opensearch-build/pull/1410#discussion_r778799905) earlier.  It brings upload to s3 component from all places to one place. I think it's good to have as it trivializes the upload to s3 for anyone to use.

We can always extend this to be able to take in bucket as a parameter as well for future use-cases. Or, I can do it now as well if that's already a use-case.